[Broker] Modify return result of NamespacesBase#internalGetBookieAffinityGroup

yuruguo opened the pull request:
### Motivation
~~Currently, there are some namespace operations that only `superUser` can access. In fact, `tenantAdmin` of this namespace should also be able to access.~~
~~The namespace operations involved includes:~~
- ~~set-bookie-affinity-group~~
- ~~get-bookie-affinity-group~~
- ~~split-namespace-bundle~~
- ~~unload-namespace-bundle~~

It is similar to this [PR](https://github.com/apache/pulsar/pull/13237#issue-1077409702), we should return `null` instead of `RestException`.

### Documentation  
- [x] `no-need-doc`

yuruguo commented:
@eolivelli @michaeljmarshall @codelipenghui @315157973 PTAL, thx!

yuruguo commented on pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:
```diff
@@ -999,12 +999,9 @@ protected BookieAffinityGroupData internalGetBookieAffinityGroup() {
                     .orElseThrow(() -> new RestException(Status.NOT_FOUND,
                             "Namespace local-policies does not exist")).bookieAffinityGroup;
-            if (bookkeeperAffinityGroup == null) {
-                throw new RestException(Status.NOT_FOUND, "bookie-affinity group does not exist");
```
It is similar to this [PR](https://github.com/apache/pulsar/pull/13237#issue-1077409702)

yuruguo commented:
/pulsarbot run-failure-checks

yuruguo commented:
/pulsarbot run-failure-checks

yuruguo commented:
/pulsarbot run-failure-checks

yuruguo commented:
/pulsarbot run-failure-checks

eolivelli reviewed (approved):
+1

massakam commented:
Should we grant these privileges to tenant administrators? At least in our company, we don't want tenant administrators to perform these operations on their own...

eolivelli reviewed (changes requested):
> Should we grant these privileges to tenant administrators? At least in our company, we don't want tenant administrators to perform these operations on their own...

My understanding is that we are not reducing the power of current users, but we are allowing tenant Admins to perform more operations.

so you are saying that you do not want tenant admin to admin tenant level operations ?

I believe that you are correct, these operations, like managing bookies are not something you want a tenant admin to be able to manage.
also unloading things
it is very dangerous 

thanks for pointing it out

I am changing my position to -1 on this patch

massakam commented:
> so you are saying that you do not want tenant admin to admin tenant level operations ?

Yes, I think these operations should only be possible by the administrators of the entire Pulsar instance, i.e. the superuser.

yuruguo commented:
@massakam @eolivelli Thanks for your reply and checking, I have reverted the modification about `authz` part, and only keep [`fix return value in method internalGetBookieAffinityGroup`](https://github.com/apache/pulsar/pull/13393#discussion_r772068978)  part, in addition, I have also changed the title

PTAL again, thx!

yuruguo commented:
/pulsarbot run-failure-checks

eolivelli commented:
@yuruguo 
can you please close this PR and open a new one ?
otherwise comments are mixed and it is hard to understand the PR in the future for people who will try to understand this work

yuruguo commented:
/pulsarbot run-failure-checks

> @yuruguo can you please close this PR and open a new one ? otherwise comments are mixed and it is hard to understand the PR in the future for people who will try to understand this work

Okay :)
